modules/mockk-agent-android-dispatcher/build.gradle.kts

```diff
@@ -3,15 +3,24 @@ import buildsrc.config.asProvider
 import com.android.build.gradle.internal.tasks.DexMergingTask
 
 plugins {
-    buildsrc.convention.`android-application`
+    id("com.android.application")
```

Comment:
I removed `android-application.gradle.kts` since it is not used except for `android-dispatcher` module.

Reply:
IMHO, I think it is better to remove `android-application.gradle.kts`.
`:mockk-agent-android-dispatcher` module is a very specific Android Application module that only extracts the final DEX file, and the build configuration is completely independent and non-reusable.
If we were to add an Android sample application module in the future and try to reuse the configuration with `:mockk-agent-android-dispatcher` module, it would add an unnecessary dependency to `:mockk-agent-android-dispatcher` module and cause the same problems that encountered in this PR.

Reply:
Yeah, I think the dependency on `:mockk-agent-android-dispatcher` should be removed from the buildSrc script